kolibri/core/auth/api.py

```diff
@@ -571,6 +572,29 @@ def create(self, request):
             return Response(serialized_user.data, status=status.HTTP_201_CREATED)
 
 
+@method_decorator(csrf_exempt, name="dispatch")
+class PublicSignUpViewSet(SignUpViewSet):
```

Comment:
I was going to comment that we should add verification here that "allow signups" is enabled on the facility, but then saw it was inherited... but then saw that we don't seem to do that validation on `SignUpViewSet` either? Does that setting only impact what shows up in the UI...??

The sending of the hashed password could be avoided by using a temporary password and then overwriting the hashed password once the user has been synced onto the local device, but if we're adding this new endpoint anyway, then I guess we might as well. And not creating the endpoint isn't ideal, as the other thing that's different here is the CSRF exemption (which removes the need for additional requests and code). And this being a public API endpoint means we'll keep it consistent over time. Though... we have it inheriting the `extract_request_data` method from the non-public viewset -- might be safest to have it just do username and password, and then set the other stuff after the sync has happened, so the API interface for this isn't changing over time.

Reply:
I'm thinking more about the sending of the hashed password... in order to do the subsequent CSR request to sync the newly created user over to the client, one will need to know the password of the user (for cases where we don't know the admin password). And if we're only working with the hashed password, we won't know the plaintext password, and the end user won't necessarily be in the loop for this process (e.g. for S2S reprovisioning), so we can't just ask them. That means the idea of using a temporary password during creation, then overwriting the hashed password locally after the sync has happened, may be the way to go?

(Side note: After peeking at the parent viewset, I was amused that if you sign up to Kolibri using the password "NOT_SPECIFIED", it creates the account but then errors out with a 500 trying to auto-sign you in...)

Reply:
(if we did end up keeping `password_hashed`, I note that I don't think I saw it being tested explicitly in the test suite)

Reply:
This is what prompted my question here:
> Should we be stopping signups in the public and private viewsets when the facility doesn't allow signups?